Add files via upload

syarikim opened the pull request:
Railsで質問・回答アプリ（エンドポイント・コントローラ設計編）課題

特に確認して頂きたいところ、自信がないところ
・質問一覧を表示する（未解決＆解決済み)のアクションはunsolvedとsolvedで自然かどうか。
・回答する　はquestions#answersとしたかが、answersコントローラでcreate にしたほうがよい？

syarikim committed:
Add files via upload

DaichiSaito commented on controller.md:
```diff
@@ -0,0 +1,20 @@
+|:----|:----|:----|:----|
+|ユーザー登録画面を表示する|GET|/users/new|users#new|
+|ユーザー登録をする|POST|/users|users#create|
+|ログイン画面を表示する|GET|/users/login|login#new|
```
`login`といったらユーザーのログインであることが明白なので`/login`で良いと思います。管理画面へのログイン画面と明確に分けるためにあえてそうしてるのかもしれませんが普通はそこまでしないかなと思います。

そして考え方的には「sessionというリソースを作る」と考えるのがRails界隈のスタンダードになってると思います。
なので`GET /login sessions#new`のほうが自然に感じます。`POST`に関しても同様です。
adminの方も同様です。

syarikim replied:
>loginといったらユーザーのログインであることが明白なので/loginで良いと思います。管理画面へのログイン画面と明確に分けるためにあえてそうしてるのかもしれませんが普通はそこまでしないかなと思います。

なるほど、分かりました。

>そして考え方的には「sessionというリソースを作る」と考えるのがRails界隈のスタンダードになってると思います。
なのでGET /login sessions#newのほうが自然に感じます。POSTに関しても同様です。
adminの方も同様です。

ここは、あまりわかってないので勉強してみます。

miketa-webprgr replied:
>「sessionというリソースを作る」

よくありがちなusersコントローラと同じように考える、っていう感じで理解するとよいかと！
usersコントローラでは、usersというリソース（資源）を作成したり、更新したり、削除したりしますが、それと同様に考えていこうという発想ですね！

「loginと同時にsessionを作成し、logoutと同時にsessionを削除するので、sessionsというリソースがあると捉えれば、usersコントローラと同じ様な`resources`パターンにおさまるし、Rails的な規約に則ってれるからいいよね！」という感じです！

DaichiSaito commented on controller.md:
```diff
@@ -0,0 +1,20 @@
+|ログインする|POST|/users/login|login#create|
+|質問一覧を表示する（全て）|GET|/questions|questions#index|
+|質問一覧を表示する（未解決）|GET|/questions/unsolved|questions#unsolved|
+|質問一覧を表示する（解決済み）|GET|/questions/solved|questions#solved|
```
これで大丈夫です 👍

DaichiSaito commented on controller.md:
```diff
@@ -0,0 +1,20 @@
+|質問詳細を表示する|GET|/questions/:id|questions#show|
+|質問編集ページを表示する|PATCH|/questions/:id/edit|questions#edit|
+|質問を削除する|DELETE|/questions/:id|questions#destroy|
+|回答する|POST|/questions/:id/answers|questions#answers|
```
コントローラとアクションがおかしいですね。
questionsコントローラで制御しようとしてるのがおかしいです。

参照
https://railsguides.jp/routing.html#%E3%83%8D%E3%82%B9%E3%83%88%E3%81%97%E3%81%9F%E3%83%AA%E3%82%BD%E3%83%BC%E3%82%B9

syarikim replied:
ここは迷ったところでした。
answers#createに直します。

DaichiSaito reviewed (commented):
コメントしました！

syarikim committed:
Update controller.md

更新箇所めも
・ログイン系のところのエンドポイントを修正
・ログイン系のところのコントローラをsessions使うように修正
・回答するのコントローラーアクションを修正

syarikim commented:
修正しました

更新箇所めも
・ログイン系のところのエンドポイントを修正
・ログイン系のところのコントローラをsessions使うように修正
・回答するのコントローラーアクションを修正

DaichiSaito commented on controller.md:
```diff
@@ -0,0 +1,20 @@
+|質問編集ページを表示する|PATCH|/questions/:id/edit|questions#edit|
+|質問を削除する|DELETE|/questions/:id|questions#destroy|
+|回答する|POST|/questions/:id/answers|answers#create|
+|ユーザー一覧を表示する|GET|/admin/users/new|admin#new|
```
ここよくみたらおかしいですね。

syarikim replied:
ありがとうございます！
おかしいですね。修正します。

syarikim committed:
Update controller.md

ユーザー一覧を表示する

syarikim commented:
更新メモ
ユーザー一覧を表示するのエンドポイントとアクションを修正

DaichiSaito reviewed (approved):
LGTM！！
